eisbuk-admin/src/components/atoms/DateNavigation/__tests__/DateNavigationOther.test.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * @jest-environment jsdom-sixteen
+ */
+
+import React from "react";
+import { screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import Button from "@material-ui/core/Button";
+
+import DateNavigation from "../DateNavigation";
+
+import { __toggleId__ } from "../__testData__/testData";
+
+import { renderWithRouter } from "@/__testUtils__/wrappers";
+
+describe("Date Navigation", () => {
+  describe("Test toggle button functionality", () => {
+    test("should not render toggle button if 'showToggle = false'", () => {
+      renderWithRouter(<DateNavigation />);
+      expect(screen.queryByTestId(__toggleId__)).toEqual(null);
+    });
+
+    test("should pass toggle state to render function", async () => {
+      const toggleTrue = "Toggle: true";
+      const toggleFalse = "Toggle: false";
+      renderWithRouter(
+        <DateNavigation showToggle>
+          {({ toggleState }) => (toggleState ? toggleTrue : toggleFalse)}
+        </DateNavigation>
+      );
+      screen.getByText(toggleFalse);
+      // this isn't so straight forward, this "checkbox" is the toggle button
```

Comment:
I think we should only test things easy to test in this context.

I'd like to add some cypress test that exercise the whole application.
Cypress can [also be used to consume stories and run tests on them](https://cmc.jazkarta.com/members/silvio-tomatis/member-purchase-receipt?key=2021:8:24:10:32:59:319622).

Reply:
Yeah...I wouldn't normally do this, but since it's a render prop, this additional test doesn't compromise the internal structure of the component. I agree that this is testing of internal state, which isn't preferred by RTL approach.

Sweet on the cypress + stories.

Will remove this.

Reply:
Hey, on second thought, I would like to keep the tests ensuring that `toggleState` gets passed to the render function as it's part of core functionality. 
The not-so-straightforward solution `getByRole()` is actually a perfectly accepted method of testing with RTL. The comment is there to provide the reader of the code with explanation as I (due to the weird bug) wasn't able to explicitly (and descriptively) test with `getByTestId()`.
If you wan't me to remove some of these tests, please specify which and I will do so.

Reply:
Or maybe just remove the state change (`.click()`) action part, but leave the test?

Reply:
Let's leave it as is for now. I just wanted to share my thoughts about how I believe this should be ideally done, but let's not take care of it right now.